pkg/testing/v1/service.go

```diff
@@ -232,6 +232,22 @@ func WithServiceAnnotations(annotations map[string]string) ServiceOption {
 	}
 }
 
+// WithConfigAnnotations assigns config annotations to a service
+func WithConfigAnnotations(annotations map[string]string) ServiceOption {
+	return func(service *v1.Service) {
+		service.Spec.ConfigurationSpec.Template.ObjectMeta.Annotations = kmeta.UnionMaps(
+			service.Spec.ConfigurationSpec.Template.ObjectMeta.Annotations, annotations)
```

Comment:
`ConfigurationSpec` and `ObjectMeta` are inlined in the struct definition so you should be able to omit those 

```suggestion
		service.Spec.Template.Annotations = kmeta.UnionMaps(
			service.Spec.Template.Annotations, annotations)
```